staging/vhost-device-gpu/src/vhu_gpu.rs

```diff
@@ -394,28 +374,29 @@ impl VhostUserGpuBackend {
         writer: &mut Writer,
         signal_used_queue: &mut bool,
     ) -> Result<()> {
-        let mut response = Err(ErrUnspec);
-
-        let (ctrl_hdr, gpu_cmd) = match GpuCommand::decode(reader) {
-            Ok((ctrl_hdr, ctrl_cmd)) => {
-                response = self.process_gpu_command(virtio_gpu, ctrl_hdr, ctrl_cmd);
-                (Some(ctrl_hdr), Some(ctrl_cmd))
+        let mut response = ErrUnspec;
+
+        let ctrl_hdr = match GpuCommand::decode(reader) {
+            Ok((ctrl_hdr, gpu_cmd)) => {
+                // TODO: consider having a method that return &'static str for logging purpose
+                let cmd_name = format!("{:?}", gpu_cmd);
+                let response_result = self.process_gpu_command(virtio_gpu, ctrl_hdr, gpu_cmd);
+                // Unwrap the response from inside Result and log information
+                response = match response_result {
+                    Ok(response) => {
+                        trace!("GpuCommand {cmd_name} success: {response:?}");
+                        response
+                    }
+                    Err(response) => {
+                        debug!("GpuCommand {cmd_name} failed: {response:?}");
+                        response
+                    }
+                };
+                Some(ctrl_hdr)
```

Comment:
Since `ctrl_hdr` is already bound in the match expression, there's no need to wrap it in `Some`. so just directly return it as the result of the match expression.

Reply:
I am returning None in the other branch of the match, that is why there needs to be Some(ctrl_hdr) here.

Reply:
this was in reference to previous comment we can leave it as it is now.